good-enough-practices-for-scientific-computing.tex

```diff
@@ -1284,26 +1341,77 @@ \subsection*{Supplementary Materials}
 people who \emph{haven't} already mastered the tools, and
 \emph{don't} realize (yet) that they will save time if and when their
 project grows.  In that situation, advocating unit testing with
-coverage checking and continuous integration is more likely to scare
-newcomers off than to aid them.
+coverage checking and continuous integration is more likely to be overwhelming
+to newcomers rather than aid them.
 
 \section*{Conclusion}
 
 We have outlined a series of practices for scientific computing based
 on our collective experience, and the experience of the thousands of
 researchers we have met through Software Carpentry, Data Carpentry,
 and similar organizations.  These practices are pragmatic, accessible
-to people who consider themselves computing novices, and can be applied
+to people who consider themselves new to computing, and can be applied
 by both individuals and groups.
 Most importantly, these practices make researchers more productive
 individually by enabling them to get more done in less time and with
 less pain.  They also accelerate research as a whole by making
 computational work (which increasingly means \emph{all} work) more
 reproducible.
 
-But progress will not happen by itself.  Universities and funding
-agencies need to support training for researchers in the use of these
-tools.  Such investment will improve confidence in the results of
+But progress will not happen by itself. The practices are being 
+increasingly incentivized through requirements from funding
+agencies and journals, but the time and skills required to put these
+approaches into practice are still not being valued. 
+
+At a local level,
+PIs can have the most impact, requiring that the research their lab
+produces follow these recommendations. Even if a PI doesn't have
+a background in computation, they can require that students show
+and share their code in lab meetings and with lab mates, that data
+is available and accessible to all in the lab and that computational
+methods sections are comprehensive. Best practices can include
+things like paired-programming and having other lab members replicate
+workflows.\ref{computing_workflows_for_biologists}. PIs can also value the time
```

Comment:
replace full stop after "workflows" with "~" to get single non-breaking space; add word "Section" before section reference.